devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -2281,6 +2342,41 @@ void YarpRobotLoggerDevice::run()
         }
     }
 
+    // Image signals are not streamed in RT
+    for (auto& [name, signal] : m_imageSignals)
+    {
+        if (!signal.connected)
+        {
+            continue;
+        }
+
+        std::lock_guard<std::mutex> lock(signal.mutex);
+        yarp::sig::ImageOf<yarp::sig::PixelRgb>* yarpImage = signal.port.read(false);
+
+        if (yarpImage != nullptr)
+        {
+            // Convert the frame from yarp to cv
+            auto colorImg = cv::Mat(yarpImage->height(),
+                                    yarpImage->width(),
+                                    yarp::cv::type_code<yarp::sig::PixelRgb>::value,
+                                    yarpImage->getRawImage(),
+                                    yarpImage->getRowSize());
+
+            // Save the frame
+            const std::filesystem::path imgPath
+                = m_exogenousImageWriters[signal.signalName].rgb->framesPath
+                  / ("img_"
+                     + std::to_string(m_exogenousImageWriters[signal.signalName].frameIndex++)
+                     + ".png");
+            cv::imwrite(imgPath.string(), colorImg);
```

Comment:
One thing I am afraid of is that this can potentially be blocking for the entire logger, especially if the image is of large dimension. Maybe it is worth using a separate thread also for this kind of signals, as for the other images?